syntax = "proto3";

// /usr/local/qqwebsrv/python/bin/python -m grpc_tools.protoc -I./ --python_out=. --grpc_python_out=. ./audio.proto

package audio;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc audio_classify (VideoRequest) returns (VideoReply) {}
}

message VideoRequest {
  string vid = 1;
  nplist mage_fp_list = 2;
  int32 max_length = 3;
  int32 detect_type = 4;
}

message nplist{
   bytes body = 1;
   npshape shape =2;
   string dtype = 3;
}

message npshape {
    int32 x = 1;
    int32 y = 2;
}

message VideoReply{
    nplist score_list = 1;
}
